Build failure: rocmPackages.rocblas

### Nixpkgs version

- Unstable (25.05)

### Steps to reproduce

1. ```nix build nixpkgs#rocmPackages.rocblas```

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/286399058

### Relevant log output

```console
# Writing Custom CMake
# Writing Kernels...
Generating kernels: Launching 12 threads...
joblib.externals.loky.process_executor._RemoteTraceback: 
"""
Traceback (most recent call last):
  File "/nix/store/5983z0gcij82c0p8k5zx5wb93am72nzs-python3.12-joblib-1.4.2/lib/python3.12/site-packages/joblib/externals/loky/process_executor.py", line 463, in _process_worker
    r = call_item()
        ^^^^^^^^^^^
  File "/nix/store/5983z0gcij82c0p8k5zx5wb93am72nzs-python3.12-joblib-1.4.2/lib/python3.12/site-packages/joblib/externals/loky/process_executor.py", line 291, in __call__
    return self.fn(*self.args, **self.kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/5983z0gcij82c0p8k5zx5wb93am72nzs-python3.12-joblib-1.4.2/lib/python3.12/site-packages/joblib/parallel.py", line 598, in __call__
    return [func(*args, **kwargs)
            ^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/3g0qsjvf88rw5bq4faih9irnzvzb16gc-python3.12-tensile-6.0.2/lib/python3.12/site-packages/Tensile/Parallel.py", line 53, in pcallWithGlobalParamsMultiArg
    return f(*args)
           ^^^^^^^^
  File "/nix/store/3g0qsjvf88rw5bq4faih9irnzvzb16gc-python3.12-tensile-6.0.2/lib/python3.12/site-packages/Tensile/TensileCreateLibrary.py", line 67, in processKernelSource
    (err, src) = kernelWriter.getSourceFileString(kernel)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/3g0qsjvf88rw5bq4faih9irnzvzb16gc-python3.12-tensile-6.0.2/lib/python3.12/site-packages/Tensile/KernelWriter.py", line 5386, in getSourceFileString
    self.getSingleCodeObjectFile(kernel)
  File "/nix/store/3g0qsjvf88rw5bq4faih9irnzvzb16gc-python3.12-tensile-6.0.2/lib/python3.12/site-packages/Tensile/KernelWriter.py", line 5317, in getSingleCodeObjectFile
    objectFileName = self.getAssembledKernelObjectFile(kernel)
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/3g0qsjvf88rw5bq4faih9irnzvzb16gc-python3.12-tensile-6.0.2/lib/python3.12/site-packages/Tensile/KernelWriter.py", line 5297, in getAssembledKernelObjectFile
    assemblyFileName = self.getKernelObjectAssemblyFile(kernel)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/3g0qsjvf88rw5bq4faih9irnzvzb16gc-python3.12-tensile-6.0.2/lib/python3.12/site-packages/Tensile/KernelWriter.py", line 5286, in getKernelObjectAssemblyFile
    kernelSource = self.getKernelSource(kernel)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/3g0qsjvf88rw5bq4faih9irnzvzb16gc-python3.12-tensile-6.0.2/lib/python3.12/site-packages/Tensile/KernelWriter.py", line 5139, in getKernelSource
    self.initKernel(kernel, tensorParametersA, tensorParametersB )
  File "/nix/store/3g0qsjvf88rw5bq4faih9irnzvzb16gc-python3.12-tensile-6.0.2/lib/python3.12/site-packages/Tensile/KernelWriterAssembly.py", line 694, in initKernel
    super(KernelWriterAssembly, self).initKernel(kernel, tPA, tPB)
  File "/nix/store/3g0qsjvf88rw5bq4faih9irnzvzb16gc-python3.12-tensile-6.0.2/lib/python3.12/site-packages/Tensile/KernelWriter.py", line 3833, in initKernel
    self.VectorWidthB = kernel["VectorWidthB"]
                        ~~~~~~^^^^^^^^^^^^^^^^
  File "/nix/store/3g0qsjvf88rw5bq4faih9irnzvzb16gc-python3.12-tensile-6.0.2/lib/python3.12/site-packages/Tensile/SolutionStructs.py", line 4629, in __getitem__
    return self._state[key]
           ~~~~~~~~~~~^^^^^
KeyError: 'VectorWidthB'
"""

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/nix/store/3g0qsjvf88rw5bq4faih9irnzvzb16gc-python3.12-tensile-6.0.2/lib/python3.12/site-packages/Tensile/bin/TensileCreateLibrary", line 43, in <module>
    TensileCreateLibrary()
  File "/nix/store/3g0qsjvf88rw5bq4faih9irnzvzb16gc-python3.12-tensile-6.0.2/lib/python3.12/site-packages/Tensile/TensileCreateLibrary.py", line 1366, in TensileCreateLibrary
    codeObjectFiles = writeSolutionsAndKernels(outputPath, CxxCompiler, None, solutions,
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/3g0qsjvf88rw5bq4faih9irnzvzb16gc-python3.12-tensile-6.0.2/lib/python3.12/site-packages/Tensile/TensileCreateLibrary.py", line 524, in writeSolutionsAndKernels
    results = Common.ParallelMap(processKernelSource, kIter, "Generating kernels")
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/3g0qsjvf88rw5bq4faih9irnzvzb16gc-python3.12-tensile-6.0.2/lib/python3.12/site-packages/Tensile/Parallel.py", line 87, in ParallelMap
    rv = Parallel(n_jobs=threadCount)(delayed(pcall)(function, a, params) for a, params in pargs)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/5983z0gcij82c0p8k5zx5wb93am72nzs-python3.12-joblib-1.4.2/lib/python3.12/site-packages/joblib/parallel.py", line 2007, in __call__
    return output if self.return_generator else list(output)
                                                ^^^^^^^^^^^^
  File "/nix/store/5983z0gcij82c0p8k5zx5wb93am72nzs-python3.12-joblib-1.4.2/lib/python3.12/site-packages/joblib/parallel.py", line 1650, in _get_outputs
    yield from self._retrieve()
  File "/nix/store/5983z0gcij82c0p8k5zx5wb93am72nzs-python3.12-joblib-1.4.2/lib/python3.12/site-packages/joblib/parallel.py", line 1754, in _retrieve
    self._raise_error_fast()
  File "/nix/store/5983z0gcij82c0p8k5zx5wb93am72nzs-python3.12-joblib-1.4.2/lib/python3.12/site-packages/joblib/parallel.py", line 1789, in _raise_error_fast
    error_job.get_result(self.timeout)
  File "/nix/store/5983z0gcij82c0p8k5zx5wb93am72nzs-python3.12-joblib-1.4.2/lib/python3.12/site-packages/joblib/parallel.py", line 745, in get_result
    return self._return_or_raise()
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/5983z0gcij82c0p8k5zx5wb93am72nzs-python3.12-joblib-1.4.2/lib/python3.12/site-packages/joblib/parallel.py", line 763, in _return_or_raise
    raise self._result
KeyError: 'VectorWidthB'
make[2]: *** [library/src/CMakeFiles/TENSILE_LIBRARY_TARGET.dir/build.make:94: Tensile/library/TensileLibrary.dat] Error 1
make[1]: *** [CMakeFiles/Makefile2:270: library/src/CMakeFiles/TENSILE_LIBRARY_TARGET.dir/all] Error 2
make: *** [Makefile:156: all] Error 2
```

### Additional context

_No response_

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.13.0, NixOS, 25.05 (Warbler), 25.05.20250124.825479c`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.24.11`
 - nixpkgs: `/nix/store/kmns8raz2za5ra55bnbj7srxrya3bild-source`

### Notify maintainers

@Madouura @Flakebi @mschwaig 

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

I believe I'm having the same issue, is that a workaround I can use for now?

```
The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/nix/store/3g0qsjvf88rw5bq4faih9irnzvzb16gc-python3.12-tensile-6.0.2/lib/python3.12/site-packages/Tensile/bin/TensileCreateLibrary", line 43, in <module>
    TensileCreateLibrary()
  File "/nix/store/3g0qsjvf88rw5bq4faih9irnzvzb16gc-python3.12-tensile-6.0.2/lib/python3.12/site-packages/Tensile/TensileCreateLibrary.py", line 1366, in TensileCreateLibrary
    codeObjectFiles = writeSolutionsAndKernels(outputPath, CxxCompiler, None, solutions,
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/3g0qsjvf88rw5bq4faih9irnzvzb16gc-python3.12-tensile-6.0.2/lib/python3.12/site-packages/Tensile/TensileCreateLibrary.py", line 524, in writeSolutionsAndKernels
    results = Common.ParallelMap(processKernelSource, kIter, "Generating kernels")
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/3g0qsjvf88rw5bq4faih9irnzvzb16gc-python3.12-tensile-6.0.2/lib/python3.12/site-packages/Tensile/Parallel.py", line 87, in ParallelMap
    rv = Parallel(n_jobs=threadCount)(delayed(pcall)(function, a, params) for a, params in pargs)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/5983z0gcij82c0p8k5zx5wb93am72nzs-python3.12-joblib-1.4.2/lib/python3.12/site-packages/joblib/parallel.py", line 2007, in __call__
    return output if self.return_generator else list(output)
                                                ^^^^^^^^^^^^
  File "/nix/store/5983z0gcij82c0p8k5zx5wb93am72nzs-python3.12-joblib-1.4.2/lib/python3.12/site-packages/joblib/parallel.py", line 1650, in _get_outputs
    yield from self._retrieve()
  File "/nix/store/5983z0gcij82c0p8k5zx5wb93am72nzs-python3.12-joblib-1.4.2/lib/python3.12/site-packages/joblib/parallel.py", line 1754, in _retrieve
    self._raise_error_fast()
  File "/nix/store/5983z0gcij82c0p8k5zx5wb93am72nzs-python3.12-joblib-1.4.2/lib/python3.12/site-packages/joblib/parallel.py", line 1789, in _raise_error_fast
    error_job.get_result(self.timeout)
  File "/nix/store/5983z0gcij82c0p8k5zx5wb93am72nzs-python3.12-joblib-1.4.2/lib/python3.12/site-packages/joblib/parallel.py", line 745, in get_result
    return self._return_or_raise()
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/5983z0gcij82c0p8k5zx5wb93am72nzs-python3.12-joblib-1.4.2/lib/python3.12/site-packages/joblib/parallel.py", line 763, in _return_or_raise
    raise self._result
KeyError: 'VectorWidthB'
make[2]: *** [library/src/CMakeFiles/TENSILE_LIBRARY_TARGET.dir/build.make:94: Tensile/library/TensileLibrary.dat] Error 1
make[1]: *** [CMakeFiles/Makefile2:270: library/src/CMakeFiles/TENSILE_LIBRARY_TARGET.dir/all] Error 2
make: *** [Makefile:156: all] Error 2
```

--- a/pkgs/development/rocm-modules/5/llvm/stage-2/1000-libcxx-failing-tests.list
+++ b/pkgs/development/rocm-modules/5/llvm/stage-2/1000-libcxx-failing-tests.list
@@ -173,3 +173,4 @@
 ../libcxx/test/std/language.support/support.dynamic/new.delete/new.delete.array/sized_delete_array14.pass.cpp
 ../libcxx/test/std/language.support/support.dynamic/new.delete/new.delete.single/sized_delete14.pass.cpp
 ../libcxx/test/libcxx/selftest/sh.cpp/empty.sh.cpp
+../libcxx/test/libcxx/transitive_includes.sh.cpp

--- a/pkgs/development/rocm-modules/5/tensile/default.nix
+++ b/pkgs/development/rocm-modules/5/tensile/default.nix
@@ -12,6 +12,7 @@
   joblib,
   filelock,
   rocminfo,
+  writeText,
 }:
 
 buildPythonPackage rec {
@@ -45,6 +46,12 @@ buildPythonPackage rec {
 
   preCheck = ''
     export ROCM_PATH=${rocminfo}
+  '';
+
+  # TODO: remove this workaround once https://github.com/NixOS/nixpkgs/pull/323869
+  # does not cause issues anymore, or at least replace it with a better workaround
+  setupHook = writeText "setup-hook" ''
+    export TENSILE_ROCM_ASSEMBLER_PATH="${stdenv.cc.cc}/bin/clang++";
   '';
 
   pythonImportsCheck = [ "Tensile" ];

--- a/pkgs/development/rocm-modules/6/tensile/default.nix
+++ b/pkgs/development/rocm-modules/6/tensile/default.nix
@@ -13,6 +13,7 @@
   joblib,
   filelock,
   rocminfo,
+  writeText,
 }:
 
 buildPythonPackage rec {
@@ -60,6 +61,12 @@ buildPythonPackage rec {
   env = {
     ROCM_PATH = rocminfo;
   };
+
+  # TODO: remove this workaround once https://github.com/NixOS/nixpkgs/pull/323869
+  # does not cause issues anymore, or at least replace it with a better workaround
+  setupHook = writeText "setup-hook" ''
+    export TENSILE_ROCM_ASSEMBLER_PATH="${stdenv.cc.cc}/bin/clang++";
+  '';
 
   pythonImportsCheck = [ "Tensile" ];
 

